src/api.v2/events/handleWorkRequest.js

```diff
@@ -21,10 +21,10 @@ const getSignedUrlIfAvailable = async (experimentId, ETag) => {
 
 
 const handleWorkRequest = async (Authorization, data) => {
-  const { experimentId } = data;
+  const { experimentId, body, requestProps } = data;
 
   // 1. Generate ETag for the new work request
-  const ETag = await generateETag(data);
+  const ETag = await generateETag({ experimentId, body, requestProps });
```

Comment:
I am not passing everything because `data` includes things like the `timeout` which shouldn't be included in the work request. Right now, I believe the fact that we are passing timeout  to the etag generation is basically invalidating the cache of every work request.